subgraphs/maker-governance/protocols/maker-governance/src/ds-chief.ts

```diff
@@ -1,85 +1,47 @@
 import { BigInt, Bytes, Address, ethereum, log } from "@graphprotocol/graph-ts";
 import { LogNote, Etch } from "../../../generated/DSChief/DSChief";
-import { VoteDelegate } from "../../../generated/DSChief/VoteDelegate";
-import {
-  Slate,
-  Spell,
-  Vote,
-  Delegate,
-  DelegateAdmin,
-} from "../../../generated/schema";
-import {
-  BIGDECIMAL_ZERO,
-  BIGINT_ONE,
-  BIGINT_ZERO,
-  SpellState,
-} from "../../../src/constants";
+import { ExecutiveVote, Slate, Spell, Voter } from "../../../generated/schema";
+import { BIGINT_ONE, SpellState } from "../../../src/constants";
 import {
   addWeightToSpells,
-  createDelegateVotingPowerChange,
+  createChiefLockedMKRChange,
   createSlate,
-  getDelegate,
   getGovernanceFramework,
+  getVoter,
   hexToNumberString,
   removeWeightFromSpells,
   toDecimal,
 } from "../../../src/helpers";
-import { VoteDelegate as VoteDelegateTemplate } from "../../../generated/templates";
 
 export function handleLock(event: LogNote): void {
   const sender = event.params.guy; // guy is the sender
   const amountStr = hexToNumberString(event.params.foo.toHexString());
   const amount = BigInt.fromString(amountStr); //.foo is the amount being locked
 
-  let newDelegateCount = 0;
-  let delegate = Delegate.load(sender.toHexString());
-  if (!delegate) {
-    delegate = new Delegate(sender.toHexString());
-    delegate.isVoteDelegate = false;
-    delegate.votingPowerRaw = BIGINT_ZERO;
-    delegate.votingPower = BIGDECIMAL_ZERO;
-    delegate.delegations = [];
-    delegate.tokenHoldersRepresented = 0;
-    delegate.currentSpells = [];
-    delegate.numberVotes = 0;
-    delegate.numberPollVotes = 0;
-
-    // Check if vote delegate contract by calling chief()
-    const voteDelegate = VoteDelegate.bind(sender);
-    const res = voteDelegate.try_delegate();
-    // will revert if not a contract
-    if (!res.reverted) {
-      // Save delegate admin to identify proxy votes later
-      const delegateAdmin = new DelegateAdmin(res.value.toHexString());
-      delegateAdmin.voteDelegate = delegate.id;
-      delegateAdmin.save();
-
-      // Track this new vote delegate contract
-      VoteDelegateTemplate.create(sender);
-
-      delegate.isVoteDelegate = true;
-      newDelegateCount = 1;
-    }
-  }
+  const voter = getVoter(sender.toHexString());
 
-  // const delegateVPChange = createDelegateVotingPowerChange(
-  //   event,
-  //   delegate.votingPowerRaw,
-  //   delegate.votingPowerRaw.plus(amount),
-  //   delegate.id,
-  //   sender.toHexString()
-  // );
-  // delegateVPChange.save();
+  // Track the change of MKR locked in chief for the user
+  const chiefLockedMKRChange = createChiefLockedMKRChange(
+    event,
+    voter.mkrLockedInChiefRaw,
+    voter.mkrLockedInChiefRaw.plus(amount),
+    voter.id
```

Comment:
I recently added the sender of the tx to this function. We need to know that in order to show who delegated the MKR.

Reply:
but for that we have the MKR delegation history , Delegation, is there any case where we want to see also when the total chief MKR increased who was the origin of it?